packages/types/src/contract/typechain/captcha/types-returns/captcha.ts

```diff
@@ -34,34 +53,39 @@ export enum Error {
 	noCorrectCaptcha = 'NoCorrectCaptcha',
 	notEnoughActiveProviders = 'NotEnoughActiveProviders',
 	providerFeeTooHigh = 'ProviderFeeTooHigh',
-	commitAlreadyExists = 'CommitAlreadyExists',
-	captchaSolutionCommitmentAlreadyExists = 'CaptchaSolutionCommitmentAlreadyExists',
-	verifyFailed = 'VerifyFailed'
+	commitAlreadyExists = 'CommitAlreadyExists'
 }
 
-export enum Payee {
-	provider = 'Provider',
-	dapp = 'Dapp'
+export type Provider = {
+	status: GovernanceStatus,
+	balance: ReturnNumber,
+	fee: number,
+	payee: Payee,
+	url: Array<number>,
+	datasetId: Hash,
+	datasetIdContent: Hash
 }
 
-export enum DappPayee {
-	provider = 'Provider',
-	dapp = 'Dapp',
-	any = 'Any'
+export type Dapp = {
+	status: GovernanceStatus,
+	balance: ReturnNumber,
+	owner: AccountId,
+	payee: DappPayee
 }
 
-export enum GovernanceStatus {
-	active = 'Active',
-	inactive = 'Inactive'
+export type UserHistorySummary = {
+	correct: number,
+	incorrect: number,
+	score: number
 }
 
 export type Commit = {
 	id: Hash,
-	user: AccountId,
+	userAccount: AccountId,
```

Comment:
Looks like we're both trying to accommodate for the other part. I had changed everything in TS to dapp/provider/user to match the contract.

Reply:
let's chat on slack :)